source/common/tracing/http_tracer_impl.cc

```diff
@@ -84,10 +84,20 @@ static void addGrpcRequestTags(Span& span, const Http::RequestHeaderMap& headers
 template <class T> static void addGrpcResponseTags(Span& span, const T& headers) {
   addTagIfNotNull(span, Tracing::Tags::get().GrpcStatusCode, headers.GrpcStatus());
   addTagIfNotNull(span, Tracing::Tags::get().GrpcMessage, headers.GrpcMessage());
-  // Set error tag when status is not OK.
+  // Set error tag when Grpc status code represents an error. See
+  // https://github.com/envoyproxy/envoy/issues/18877
   absl::optional<Grpc::Status::GrpcStatus> grpc_status_code = Grpc::Common::getGrpcStatus(headers);
   if (grpc_status_code && grpc_status_code.value() != Grpc::Status::WellKnownGrpcStatus::Ok) {
```

Comment:
We should check `grpc_status_code.value() != Grpc::Status::WellKnownGrpcStatus::InvalidCode)` here rather than `!ok`

NOT_OK is redundant once you added those new checks in this PR

Reply:
Fixed.